### PR TITLE
update depdendencies for webpack 5

### PR DESCRIPTION
**Pull request by solomontang**
- update dependencies to use webpack5
- move core lattice dependencies to peerDependencies
- remove unused loaders and files

### CODE REVIEW

File: config/babel/babel.config.js [hunk 1]
```diff
@@ -2,36 +2,46 @@ module.exports = {
...
+        transform: (importName) => `@fortawesome/pro-solid-svg-icons/${importName}`,
       },
     }],
   ],
   presets: [
-    '@babel/preset-env',
+    ['@babel/preset-env', {
+      corejs: '3',
+      useBuiltIns: 'entry',
+    }],
```
Comment (UnsungHero97) [inline config/babel/babel.config.js:44]: I based changes in our other libraries on this comment:

https://github.com/babel/babel/issues/9853#issuecomment-619587386

> If you are authoring a library, use only @babel/transform-runtime with corejs option plus @babel/runtime-corejs3 as dependency, and @babel/preset-env for syntax transpilation with useBuiltIns: false.

So this should be:

```
['@babel/preset-env', {
  useBuiltIns: false,
}],
```

File: config/webpack/webpack.config.js [hunk 2]
```diff
@@ -60,43 +66,29 @@ module.exports = (env = {}) => {
...
-        commonjs2: 'react-dom',
-        commonjs: 'react-dom',
-        amd: 'react-dom'
-      },
-      'styled-components': {
-        amd: 'styled-components',
-        commonjs: 'styled-components',
-        commonjs2: 'styled-components',
-      },
-    },
```
Comment (UnsungHero97) [inline config/webpack/webpack.config.js:83]: You added `webpack-node-externals` to `package.json` but you're not using it here.

File: package.json [hunk 3]
```diff
@@ -24,93 +24,105 @@
...
-    "luxon": "~1.25.0",
     "polished": "~4.1.0",
     "prop-types": "~15.7.2",
     "react-scrolllock": "~5.0.1",
-    "react-select": "~4.0.2",
-    "react-transition-group": "~4.4.1",
-    "redux-reqseq": "0.5.0"
+    "react-select": "~4.3.1",
+    "react-transition-group": "~4.4.2",
+    "redux-reqseq": "0.6.1"
```
Comment (UnsungHero97) [inline package.json:39]: This can be moved to externals.

File: package.json [hunk 4]
```diff
@@ -24,93 +24,105 @@
...
-    "@storybook/theming": "~6.1.17",
+    "@babel/cli": "~7.14.5",
+    "@babel/core": "~7.14.5",
+    "@babel/plugin-proposal-class-properties": "~7.14.5",
+    "@babel/plugin-proposal-object-rest-spread": "~7.14.5",
+    "@babel/plugin-transform-runtime": "~7.14.5",
+    "@babel/preset-env": "~7.14.5",
+    "@babel/preset-flow": "~7.14.5",
+    "@babel/preset-react": "~7.14.5",
+    "@babel/runtime": "~7.14.5",
```
Comment (UnsungHero97) [inline package.json:50]: I don't think this is necessary because of `@babel/runtime-corejs3` in deps above.

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

**Review by UnsungHero97 — APPROVED** [review]

**Review by anzioka — APPROVED** [review]

**Pull request merged**